### DISCUSSION TITLE
How to update on conflict in MySQL?

### DISCUSSION THREAD

**Discussion by lz1998** [SeaQL/sea-orm#366] [Q&A]
unique_index conflict

**Comment by billy1624** [comment]
Ohohhh lots of questions loll

Same here, no lative support yet. I guess you have to go raw sql.
- https://www.sea-ql.org/SeaORM/docs/basic-crud/raw-sql#query-by-raw-sql

Reated: #187

**Comment by SebastienGllmt** [comment]
Looks like the upstream sea-query feature was merged 3 days ago https://github.com/SeaQL/sea-query/pull/279 and released as part of sea-query 0.24.0

**Reply by billy1624** [reply]
Hey @SebastienGllmt, yes, the upstream is ready. I was brainstorming the upsert API of SeaORM. Wonder if you have any thoughts on it?

**Comment by kenkoooo** [comment]
I just discovered that the `on_conflict` function is already available in the latest version: https://www.sea-ql.org/SeaORM/docs/basic-crud/insert/#on-conflict

**Comment by wojciechbator** [comment]
Is it possible in SeaORM to pass constraint to on_conflict? I'd like to achieve something like `ON CONFLICT ON CONSTRAINT (constraint_name)`.  The docs are somewhat lacking on this matter @billy1624 and I'm not sure how to achieve that. I think alternatively I could just craft the query using SeaQuery, but I'd prefer SeaORM if it's possible.

Upd: Alternatively, I can use current columns constraint, but suppose that I need to use normal and nullable column, handling it with expression, like so:

`ON CONFLICT (some_col1, some_col2, coalesce(nullable_col, ''))`

Is there a possible way in SeaORM / SeaQuery to pass such mix? It's doable in Postgres 14 for sure (Postgres 15+ handles nullable cols even better).

Update 2: I carefully checked SeaQuery and `COALESCE` support and found out it's totally doable like so:

```
            .on_conflict(
                OnConflict::new()
                    .exprs([
                        Expr::col(MyEntity::Column::Col1),
                        Expr::col(MyEntity::Column::Col2),
                        Expr::expr(Expr::col(MyEntity::Column::ColNullable).if_null("")),
                    ])
                    .update_columns([
```
Good job guys <3
